### PR TITLE
Update effort fetching logic as per the MFM case

### PR DESCRIPTION
**Pull request by GauravGusain98**
### Changes
- Removed hardcoded indexing for effort sheet
- If an effort sheet is having multiple projects then the effort related to each project will be associated with their corresponding projects in Employee Portal

### CODE REVIEW

File: Modules/Project/Console/SyncEffortsheet.php [hunk 1]
```diff
@@ -65,66 +67,147 @@ public function handle()
                 $sheetId = $correctedEffortsheetUrl[1];
                 $sheet = new Sheets();
                 $projectMembersCount = $project->teamMembers()->count();
-                $range = 'C2:G' . ($projectMembersCount + 1); // this will depend on the number of people on the project
+                $endColumn = config('efforttracking.default_end_column_in_effort_sheet');
```
Comment (P4NK4J) [inline Modules/Project/Console/SyncEffortsheet.php:70]: Can we make it `last_column` ?

**Review by P4NK4J — COMMENTED** [review]
Minor comment else looks good.

File: Modules/Project/Console/SyncEffortsheet.php (hunk 1 above)
Comment (GauravGusain98) [reply Modules/Project/Console/SyncEffortsheet.php]: done

File: Modules/Project/Console/SyncEffortsheet.php [hunk 2]
```diff
@@ -65,66 +67,146 @@ public function handle()
...
+                        try {
+                            $effortData['sheet_project'] = $sheetProject;
+                            $this->updateEffort($effortData);
+                        } catch (Exception $e) {
+                            continue;
+                        }
                     }
+                }
+            } catch (Exception $e) {
+                dd($e);
```
Comment (kuldeep3) [inline Modules/Project/Console/SyncEffortsheet.php:164]: 😮‍💨 💢

File: Modules/Project/Console/SyncEffortsheet.php [hunk 3]
```diff
@@ -65,66 +67,146 @@ public function handle()
                 $sheetId = $correctedEffortsheetUrl[1];
                 $sheet = new Sheets();
```
Comment (kuldeep3) [inline Modules/Project/Console/SyncEffortsheet.php:68]:
```suggestion
                $sheets = new Sheets();
```

File: Modules/Project/Console/SyncEffortsheet.php [hunk 4]
```diff
@@ -65,66 +67,146 @@ public function handle()
...
                 $projectMembersCount = $project->teamMembers()->count();
-                $range = 'C2:G' . ($projectMembersCount + 1); // this will depend on the number of people on the project
+                $lastColumn = config('efforttracking.default_last_column_in_effort_sheet');
+                $columnIndex = 5;
+                $projectsInSheet = [];
 
+                try {
+                    while (true) {
+                        $range = 'C1:' . ++$lastColumn . '1';
+                        $sheets = $sheet->spreadsheet($sheetId)
```
Comment (kuldeep3) [inline Modules/Project/Console/SyncEffortsheet.php:77]:
```suggestion
                        $sheet = $sheets->spreadsheet($sheetId)
```

Comment (kuldeep3) [reply Modules/Project/Console/SyncEffortsheet.php:77]: Please update at other places too. ⚒️

**Review by kuldeep3 — APPROVED** [review]
Few comments, rest of all looks good. Please self-review once before assigning it for review.

File: Modules/EffortTracking/Config/config.php [hunk 5]
```diff
@@ -4,4 +4,13 @@
     'name' => 'EffortTracking',
     'minimum_expected_hours' => 8,
     'update_date_count_after_time' => '18:00:00',
+    'columns_name' => [
+        'team_member_name' => 'individual effort',
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/EffortTracking/Config/config.php:8]: Ohh...I think we should rename it. Actually this name is what we are using in the effortsheets. So we also need to update this name in every effortsheet as well.

[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/EffortTracking/Config/config.php:8]: Changed the name

File: Modules/Project/Console/SyncEffortsheet.php (hunk 2 above)
Comment (GauravGusain98) [reply Modules/Project/Console/SyncEffortsheet.php:164]: corrected

File: Modules/Project/Console/SyncEffortsheet.php (hunk 3 above)
Comment (GauravGusain98) [reply Modules/Project/Console/SyncEffortsheet.php:68]: done

File: Modules/Project/Console/SyncEffortsheet.php (hunk 4 above)
Comment (GauravGusain98) [reply Modules/Project/Console/SyncEffortsheet.php:77]: done

File: Modules/Project/Console/SyncEffortsheet.php (hunk 3 above)
Comment (kuldeep3) [reply Modules/Project/Console/SyncEffortsheet.php:68]: @GauravGusain98 Was this implemented as I cannot see this change in #1465 🤔

**Pull request merged**